### ISSUE TITLE
Vidline wants to access key "com.firebase.FiRInstallations.installations.1:19 312161103:ios:513816e35120d976bf153a" in your keychain.

### ISSUE THREAD

**Issue by Tang-Hai** [firebase/firebase-ios-sdk#9392]
How to stop this popup from appearin !!!

![34241645540838_ pic](https://user-images.githubusercontent.com/13147415/155712377-d4731d0b-db21-4d07-b086-134847db1e13.jpg)
g

**Comment by paulb777** [comment]
Do the suggestions in #8335 or #8950 help?

**Comment by Tang-Hai** [comment]
> #8335或#8950中的建议有帮助吗？

I don't understand how to modify it, are there any detailed steps to modify it?

**Comment by ncooke3** [comment]
Hi @Tang-Hai, 

Sorry for the trouble. The steps for https://github.com/firebase/firebase-ios-sdk/issues/8335 are to ensure that your project is properly signed and has the enabled the "Keychain Sharing" capability. Here's a [Stack Overflow answer](https://stackoverflow.com/questions/65131018/how-can-i-enable-the-keychain-sharing-capability-in-xcode/65132191) with instructions on how to enable keychain sharing.

If the above advice does resolve the issue, I have three follow-up questions to help the team better understand the problem:
- What version of Firebase is being used in your project?
- What Firebase products are being used?
- What platforms are you seeing this on? (macOS or macCatalyst or both?)

Thanks!

**Comment by Tang-Hai** [comment]
> 



> 嗨@唐海，
> 
> 抱歉，添麻烦了。#8335的步骤是确保您的项目已正确签名并启用了“钥匙串共享”功能。这是一个[Stack Overflow 答案](https://stackoverflow.com/questions/65131018/how-can-i-enable-the-keychain-sharing-capability-in-xcode/65132191)，其中包含有关如何启用钥匙串共享的说明。
> 
> 如果上述建议确实解决了问题，我有三个后续问题可以帮助团队更好地理解问题：
> 
> * 您的项目中使用的是哪个版本的 Firebase？
> * 正在使用哪些 Firebase 产品？
> * 你在什么平台上看到这个？（macOS 或 macCatalyst 或两者兼而有之？）
> 
> 谢谢！

It still pops up, I read in the firebase documentation that macos are supported.
I am using Analytics , Crashlytics these features.
Can you help me with the popup issue?

**Comment by Tang-Hai** [comment]
> 我有同样的问题： <img alt="弹出访问火力基地" width="433" src="https://user-images.githubusercontent.com/25382699/158633835-0054869a-9dfa-4b30-bc49-6a8d430f036c.png">
> 
> 每次用户打开应用程序时，我都会两次弹出此弹出窗口。
> 
> 我正在使用 TestFlight 分发一个 macOS 应用程序，第一次没有弹出窗口，因为我：
> 
> * 在我的项目中介绍了`kSecUseDataProtectionKeychain: true`我当前的钥匙串实现查询
> * 在 macOS.entitlements 中添加了这个定义：
> 
> ```
> 	<key>keychain-access-groups</key>
> 	<array>
> 		<string>$(AppIdentifierPrefix)*</string>
> 	</array>
> ```
> 
> MacOS 权利基本上是这样的： <img alt="截图 2022-03-16 at 17 36 55" width="411" src="https://user-images.githubusercontent.com/25382699/158641244-4f99c164-d0f9-463d-84cf-be80f1ffd0c7.png">
> 
> 但是当我发布更新时 - 没有任何代码更改，两个弹出窗口就会出现。一个用于 FIRInstallation...，另一个以我的 Bundle Identifier 命名 - 两个弹出窗口都用于授予访问钥匙串的权限。**每次用户打开应用程序时，他们都必须授予访问钥匙串的权限。**
> 
> 如果我在我的项目中删除这个`FirebaseApp.configure()`- 那么我就没有问题了，而且我在 Firebase 创建的钥匙串中看不到两个项目（一个是“FIRInstallation ...”，另一个是我的应用程序包标识符）。我可以分享两个项目之一的屏幕截图，一个名为“com.firebase.FIRInstallations ...”的项目： <img alt="钥匙串火力基地" width="965" src="https://user-images.githubusercontent.com/25382699/158636025-70506591-fda0-499c-b589-f8ee4c52b972.png">
> 
> 我想说的是，当启动 macOS 应用程序时，FirebaseApp.configure() 在用户 macOS 钥匙串中创建两个项目，在“登录”中创建两个条目，并引入两个弹出窗口来访问钥匙串中的这两个项目每次用户打开应用程序。
> 
> PS：我的钥匙串中名为我的包标识符的钥匙串登录项​​不是我项目中钥匙串实现的结果。因为，正如我所说，如果我删除 FirebaseApp.configure() 我在安装应用程序并运行它时看不到它。此外，我的钥匙串实现使用不同的名称来存储诸如 apitokens 之类的东西......
> 
> 对于@ncooke3：
> 
> > 您的项目中使用的是哪个版本的 Firebase？
> 
> 8.9.1 并尝试使用 8.11（问题仍然存在）
> 
> > 正在使用哪些 Firebase 产品？
> 
> 分析和崩溃分析
> 
> > 你在什么平台上看到这个？（macOS 或 macCatalyst 或两者兼而有之？）
> 
> 苹果系统
> 
> Firebase 团队，我们在这里需要帮助 :D

Hi, is your problem solved? Are there alternatives?

**Comment by denismullaraj** [comment]
@Tang-Hai No unfortunately I haven't been able to solve it myself. I tried everything I could the keychain access group, the kSecUseDataProtectionKeychain flag... that only seems to solve issues on my side of the Keychain Usage. Not on the Firebase library. That's the reason I'm waiting for Firebase to give us more indication or guide us how we can get rid of it.

**Comment by DesmanLead** [comment]
Is it possible to use something except the keychain on macOS to store the identifier?
We'd like to opt-out from using the keychain.

**Comment by ncooke3** [comment]
I wanted to leave a quick update here: we have been investigating the issue and are currently working on a solution. 

I will share an another update soon– thank you for everyone's patience!

**Comment by Tang-Hai** [comment]
> I wanted to leave a quick update here: we have been investigating the issue and are currently working on a solution.
> 
> I will share an another update soon– thank you for everyone's patience!

Really looking forward to your update

**Comment by tommienu** [comment]
Sorry for the bump, but do we have an update on this topic? It's currently blocking any macOS release with Firebase.

**Comment by ncooke3** [comment]
Hi everyone (and thanks for the bump @tommienu),

**tldr;** We identified a solution and are working to get everything working in Firebase 9.6.0. Currently, this release is scheduled to be released next week.  🎉 

From Firebase 9.6.0 onward, macOS targets depending on Firebase should be signed with a provisioning profile that has the Keychain Sharing capability enabled. This will be mentioned in the release notes and documentation. 

---

#### A little background behind the above requirement:
The reason this is needed is because in order to make the permission pop-ups go away, the macOS keychain has to be configured to treat items like the iOS-style keychain. The iOS-style keychain doesn't request access from the user because iOS apps have a sandbox to freely interact with their designated portion of the keychain. To replicate this behavior on macOS, we had to make a code change to tell the macOS keychain to treat interactions like that on iOS (thisis what is being released in Firebase 9.6.0).  However, this change alone is not enough as macOS needs to know that a Mac app has access to modify its own items. Only you, the developer, can give official authorization so **macOS targets that depend on Firebase** need to be signed with a provisioning profile that has the Keychain Sharing capability enabled. Together, the keychain pop-ups will go away as the system will know exactly what keychain items your Mac app has permission to modify.

And, in case it is not clear, these changes _only_ apply to macOS targets using Firebase. Non-macOS apps will not require any action.

**Comment by ncooke3** [comment]
Hi everyone, 

Just a quick fyi that Firebase 9.6.0 has been released. Please give it a go and let us know if you run into any trouble! 

As noted in [above](https://github.com/firebase/firebase-ios-sdk/issues/9392#issuecomment-1238751984) post and in the [release notes](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseCore/CHANGELOG.md#firebase-960), please enable the Keychain Sharing capability for your macOS targets to make sure everything works properly.

Thanks!
